v2/internal/plugins/collector/proxy/context.go

```diff
@@ -256,6 +260,14 @@ func (pc *PluginContext) Metrics(clear bool) []*types.Metric {
 	if clear {
 		pc.sessionMts = nil
 	}
+
+	globalPrefix := pc.ctxManager.globalPrefix
+	if globalPrefix.enabled && globalPrefix.removeFromOutput {
+		for i := 0; i < len(mts); i++ {
+			mts[i].Namespace_ = mts[i].Namespace_[1:]
```

Comment:
are we sure there's always a sufficient amount of namespace segments so that this won't be out of bounds?

Reply:
yes, 
- AddMetrics go through IsUsableForAddition which have check if len(ns.elements) < minNamespaceLength (2)